modules/rostests/apitests/user32/ScrollBarRedraw.c

```diff
@@ -0,0 +1,712 @@
+/*
+ * PROJECT:     ReactOS api tests
+ * LICENSE:     ISC (https://spdx.org/licenses/ISC)
+ * PURPOSE:     Tests window redrawing when scrollbars appear or disappear
+ * COPYRIGHT:   Copyright 2024 Marek Benc <benc.marek.elektro98@proton.me>
+ */
+
+#include <windows.h>
+#include "wine/test.h"
+
+#define TEST_CLASS_NAME   "ScrollBarRedraw"
+#define TEST_WINDOW_TITLE "ScrollBarRedraw"
+
+EXTERN_C IMAGE_DOS_HEADER __ImageBase;
+#define HINST_THISCOMPONENT ((HINSTANCE)&__ImageBase)
+
+static LRESULT CALLBACK WindowProc(HWND Window, UINT Message, WPARAM wParam, LPARAM lParam);
+
+#define TEST_COLOR_COUNT 16
+static COLORREF Colors[TEST_COLOR_COUNT] = { 0 };
+static HBRUSH   ColorBrushes[TEST_COLOR_COUNT] = { 0 };
+
+static BOOL HaveHRedraw = FALSE;
+static BOOL HaveVRedraw = FALSE;
+
+typedef enum
```

Comment:
```suggestion
typedef enum _FSM_STATE
```
to avoid nameless enums in the debugger.